File: fixtures/scale/python/large/cipherlab/checks/check_riverstream.py
from cipherlab.ciphers.riverstream import riverstream_encrypt, riverstream_decrypt
from cipherlab.specs.riverstream_spec import riverstream_spec
from cipherlab.corpus.corpus_stream import corpus_stream


def check_riverstream():
    spec = riverstream_spec()
    for text in corpus_stream():
        sealed = riverstream_encrypt(text, spec["key"])
        opened = riverstream_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "stream"

File: fixtures/scale/ruby/large/lib/check_riverstream.rb
require_relative 'riverstream_encrypt'
require_relative 'riverstream_decrypt'
require_relative 'corpus_stream'

def check_riverstream
  corpus_stream.each do |text|
    sealed = riverstream_encrypt(text, 14)
    opened = riverstream_decrypt(sealed, 14)
    return false if opened != text
  end
  true
end
